packages/gatsby-theme-carbon/src/components/ResourceCard/ResourceCard.js

```diff
@@ -1,112 +1,184 @@
-            <Error20 aria-label="disabled" />
-          ) : null}
-        </div>
-      </>
-    );
+const ResourceCard = ({
+  children,
+  href,
+  subTitle,
+  title,
+  color = 'light',
+  disabled = false,
+  aspectRatio = '2:1',
+  actionIcon = 'launch',
+  className,
+  type = 'default',
+  ...rest
+}) => {
+  let isLink;
+  if (href !== undefined) {
+    isLink = href.charAt(0) === '/';
+  }
 
-    let cardContainer;
-    if (disabled === true || href === undefined) {
-      cardContainer = <div className={carbonTileclassNames}>{cardContent}</div>;
-    } else if (isLink === true) {
-      cardContainer = (
-        <Link to={href} className={carbonTileclassNames} {...rest}>
-          {cardContent}
-        </Link>
-      );
-    } else {
-      cardContainer = (
-        <a href={href} className={carbonTileclassNames} {...rest}>
-          {cardContent}
-        </a>
-      );
-    }
-
-    return (
-      <div className={ResourceCardClassNames}>
-        <div className={aspectRatioClassNames}>
-          <div className={`${prefix}--aspect-ratio--object`}>
-            {cardContainer}
+  // This represents the variant of Resource Card: default, disabled, dark, and mini.
+  const ResourceCardClassNames = classnames(`${prefix}--resource-card`, {
+    [className]: className,
+    [`${prefix}--resource-card--disabled`]: disabled,
+    [`${prefix}--resource-card--dark`]: color === 'dark',
+    [`${prefix}--resource-card--mini`]: type === 'mini',
+  });
+
+  // This gives the Resource Card it's properties for handling links
+  const carbonTileclassNames = classnames(`${prefix}--tile`, {
+    [`${prefix}--tile--clickable`]: href !== undefined,
+  });
+
+  // These are all the icons available to the resource card.
+  const icons = (
+    <>
+      {actionIcon === 'launch' && !disabled ? (
+        <Launch20 aria-label="Open resource" />
+      ) : null}
+      {actionIcon === 'arrowRight' && !disabled ? (
+        <ArrowRight20 aria-label="Open resource" />
+      ) : null}
+      {actionIcon === 'download' && !disabled ? (
+        <Download20 aria-label="Download" />
+      ) : null}
+      {actionIcon === 'email' && !disabled ? (
+        <Email20 aria-label="Email" />
+      ) : null}
+      {actionIcon === 'calendar' && !disabled ? (
+        <Calendar20 aria-label="Calendar" />
+      ) : null}
+      {actionIcon === 'disabled' || disabled === true ? (
+        <Error20 aria-label="disabled" />
+      ) : null}
+    </>
+  );
+
+  // This determines the size of the title prop. productive-heading-03 for default and body-long-02 for Mini.
+  const titleSize = (
+    <>
+      {type === 'default'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__title`}>{title}</h4>
+          )
+        : null}
+      {type === 'mini'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__mini-title`}>{title}</h4>
+          )
+        : null}
+    </>
+  );
```

Comment:
Could we maybe simplify this? 
```suggestion
 const titleClassNames = classnames({
 [`${prefix}--resource-card__title`]: type === "default",
 [`${prefix}--resource-card__mini-title]: type === "mini",
 });
```